modules/Sources/Features/SendFlow/SendFlowStore.swift

```diff
@@ -67,47 +67,46 @@ public struct SendFlowReducer: Reducer {
         }
 
         public var isInvalidAddressFormat: Bool {
-            !transactionAddressInputState.isValidAddress
-            && !transactionAddressInputState.textFieldState.text.data.isEmpty
+            !transactionAddressInputState.textFieldState.text.data.isEmpty
+            && !transactionAddressInputState.isValidAddress
         }
 
         public var isInvalidAmountFormat: Bool {
-            !transactionAmountInputState.textFieldState.valid
-            && !transactionAmountInputState.textFieldState.text.data.isEmpty
+            !transactionAmountInputState.textFieldState.text.data.isEmpty
+            && !transactionAmountInputState.isValidInput
         }
         
         public var isValidForm: Bool {
-            @Dependency(\.zcashSDKEnvironment) var zcashSDKEnvironment
-
-            return transactionAmountInputState.amount.data > zcashSDKEnvironment.network.constants.defaultFee().amount
-            && transactionAddressInputState.isValidAddress
+            !isInvalidAddressFormat
             && !isInsufficientFunds
             && memoState.isValid
+            && !isInvalidAmountFormat
         }
-        
+
         public var isInsufficientFunds: Bool {
-            transactionAmountInputState.amount.data > transactionAmountInputState.maxValue.data
+            @Dependency(\.zcashSDKEnvironment) var zcashSDKEnvironment
+
+            return transactionAmountInputState.amount.data > spendableBalance.amount - zcashSDKEnvironment.network.constants.defaultFee().amount
```

Comment:
This will need changing for ZIP 317, which is a requirement for Zashi 1.0.

Reply:
Yes, that will be handled in a separate zip317 related ticket.